server/src/main/java/org/elasticsearch/index/mapper/FloatVectorFieldMapper.java

```diff
@@ -46,7 +49,56 @@ public static final class Defaults {
 
         private Defaults() {}
 
-        public static final FieldType FIELD_TYPE = new FieldType();
+        public static final int DIMENSION_LIMIT = 2048;
+
+        public static final FieldType FIELD_TYPE = new FieldType() {
```

Comment:
I know it looks ugly, the reason for this is that dimension is private and cannot be accessed from a subclass. The only alternative I came up with is to re-implement the whole `FieldType` which would duplicate all the code of FieldType.